File: android/src/main/java/com/azifacemobile/theme/OcrConfirmation.java
package com.azifacemobile.theme;

import android.graphics.Typeface;

import com.azifacemobile.theme.abstracts.ViewStyle;
import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONException;
import org.json.JSONObject;

import javax.annotation.Nullable;

public class OcrConfirmation extends ViewStyle {
  private static final String KEY = "ocrConfirmation";
  private final Theme theme;
  private final JSONObject target;
  private final Color color;
  private final Font font;
  private final Button button;
  private final ScrollIndicator scrollIndicator;
  private final InputField inputField;

  public OcrConfirmation(ReactApplicationContext context) {
    super(KEY);

    this.theme = new Theme();
    this.target = this.theme.getTarget(KEY);
    this.color = new Color();
    this.font = new Font(context);
    this.button = new Button(context, this.target);
    this.scrollIndicator = new ScrollIndicator(context, this.target);
    this.inputField = new InputField(context, this.target);
  }

  @Override
  public int getBackgroundColor() {
    return super.getBackgroundColor(Color.TRANSPARENT);
  }

  public boolean getEnableFixedConfirmButton() {
    final String key = "isFixedConfirmButton";
    try {
      if (!this.theme.exists(this.target, key)) {
        return false;
      }

      return this.target.getBoolean(key);
    } catch (JSONException e) {
      return false;
    }
  }

  public int getLineColor() {
    return this.color.getColor(this.target, "lineColor", "#026ff4");
  }

  public int getLineWidth() {
    final int defaultLineWidth = -1;
    final String key = "lineWidth";
    try {
      if (!this.theme.exists(this.target, key)) {
        return defaultLineWidth;
      }

      return this.target.getInt(key);
    } catch (JSONException e) {
      return defaultLineWidth;
    }
  }

  public int getHeaderTextColor() {
    return this.color.getColor(this.target, "headerTextColor", "#026ff4");
  }

  @Nullable
  public Typeface getHeaderFont() {
    return this.font.getTypography(this.target, "headerFont");
  }

  public int getSectionTextColor() {
    return this.color.getColor(this.target, "sectionTextColor", "#272937");
  }

  @Nullable
  public Typeface getSectionFont() {
    return this.font.getTypography(this.target, "sectionFont");
  }

  public Button getButton() {
    return this.button;
  }

  public ScrollIndicator getScrollIndicator() {
    return this.scrollIndicator;
  }
  public InputField getInputField() {
    return this.inputField;
  }
}

File: ios/Theme/OcrConfirmation.swift
public class OcrConfirmation: ViewStyle {
  private static let KEY: String = "ocrConfirmation"
  private let style: Style
  private let target: NSDictionary?
  private let color: Color
  private let font: Font
  private let button: Button
  private let scrollIndicator: ScrollIndicator
  private let inputField: InputField

  init() {
    self.style = Style()

    self.target = self.style.getTarget(OcrConfirmation.KEY)
    self.color = Color()
    self.font = Font()
    self.button = Button(target: self.target)
    self.scrollIndicator = ScrollIndicator(target: self.target)
    self.inputField = InputField(target: self.target)

    super.init(key: OcrConfirmation.KEY)
  }

//  TODO: Add it in the next versions because it currently is wrong on iOS
//  override public func getBackgroundColor() -> UIColor {
//    return super.getBackgroundColor(defaultColor: Color.TRANSPARENT)
//  }

  public func getEnableFixedConfirmButton() -> Bool {
    let key = "isFixedConfirmButton"
    if !self.style.exists(self.target, key: key) {
      return false
    }

    return self.target?[key] as? Bool ?? false
  }

  public func getLineColor() -> UIColor {
    return self.color.getColor(self.target, key: "lineColor", defaultColor: "#026ff4")
  }

  public func getLineWidth() -> Int32 {
    let defaultLineWidth: Int32 = -1
    let key = "lineWidth"
    if !self.style.exists(self.target, key: key) {
      return defaultLineWidth
    }

    return self.target?[key] as? Int32 ?? defaultLineWidth
  }

  public func getHeaderTextColor() -> UIColor {
    return self.color.getColor(self.target, key: "headerTextColor", defaultColor: "#026ff4")
  }

  public func getHeaderFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "headerFont")
  }

  public func getSectionTextColor() -> UIColor {
    return self.color.getColor(self.target, key: "sectionTextColor", defaultColor: "#272937")
  }

  public func getSectionFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "sectionFont")
  }

  public func getLabelColor() -> UIColor {
    return self.color.getColor(self.target, key: "labelColor", defaultColor: "#272937")
  }

  public func getLabelFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "labelFont")
  }

  public func getButton() -> Button {
    return self.button
  }

  public func getScrollIndicator() -> ScrollIndicator {
    return self.scrollIndicator
  }

  public func getInputField() -> InputField {
    return self.inputField
  }
}
